ui/qt-lib/src/Ariadne/UI/Qt/Widgets/WalletTree.hs

```diff
@@ -102,8 +102,7 @@ handleWalletTreeEvent UiLangFace{..} ev = do
   lift $ case ev of
     WalletTreeNewWalletCommandResult _commandId result -> case result of
       UiNewWalletCommandSuccess mnemonic -> do
-        void $ QMessageBox.information treeView ("Success" :: String) $
-          toString $ "This is your wallet mnemonic. Save it.\n\n" <> intercalate " " mnemonic
+        void $ runConfirmMnemonic mnemonic
```

Comment:
If user fails this dialog, the wallet is still created. In fact, it is already created by the time this dialog starts. Do we want to delete it if user can't confirm he wrote down the mnemonic? I'm not sure what UX Alexander meant with this.

Reply:
User should see that the wallet has been created only after they have entered the correct recovery phrase. And that's a problem because currently mnemonic generation and wallet creation are coupled together. We need to decouple them.
I suppose we need a new command which generates a mnemonic and returns it, without creating a wallet. And then we can call `restore` with this mnemonic and `full:false` when the mnemonic is entered correctly.